src/frontend/ocamlmerlin_server.ml

```diff
@@ -79,7 +79,9 @@ let main () =
   (* Setup env for extensions *)
   Unix.putenv "__MERLIN_MASTER_PID" (string_of_int (Unix.getpid ()));
   match List.tl (Array.to_list Sys.argv) with
-  | "single" :: args -> exit (New_merlin.run "" args)
+  | "single" :: args -> exit (New_merlin.run 
+    (String.concat "" (Array.to_list (Array.map (fun s -> s ^ "\000") (Unix.environment ())))) 
+    args)
```

Comment:
I don't think this diff is meant to be there. Seems to be copied from #848 .

Reply:
Thanks - I will remove it, but is there a fix for the related bug in the latest master?